lib/models/user-identity.js

```diff
@@ -121,9 +121,7 @@
                        loopback.getModelByType(loopback.User);
        var userObj = (options.profileToUser || profileToUser)(provider, profile, options);
        if (!userObj.email && !options.emailOptional) {
-         process.nextTick(function() {
-           return cb('email is missing from the user profile');
-         });
+         return cb('email is missing from the user profile');
```

Comment:
Thanks, I've made the change now. Just out of curiosity why do we need the callback to be called in a `nextTick()`? Since we need to return out of that function immediately if the email doesn't exist shouldn't it be fine if we just return the callback straight away?